docs/extend/creating-classic-plugins.md

```diff
@@ -41,28 +41,130 @@ These examples provide the bare bones needed to get started. For more informatio
+
+Example:
+```yaml
+org.example.module: # or 'ALL-UNNAMED' if the plugin is non-modular
+  - inbound_network
+```
+
+#### `load_native_libraries`
+Allows code to load native libraries and call restricted methods. This entitlement also enables native access for modules it is granted to (see [Restricted Methods](https://docs.oracle.com/en/java/javase/24/core/restricted-methods.html) in the Java documentation). Native code may alter the JVM or circumvent access checks such as file or network restrictions.
+
+Example:
+```yaml
+org.example.module: # or 'ALL-UNNAMED' if the plugin is non-modular
+  - load_native_libraries
+```
+
+#### `files`
+
+Allows code to access the filesystem, to read or write paths as specified by the entitlement's fields. The filesystem of the OS hosting {{es}} may contain sensitive files, for example credentials. Some files are meant to be always accessible to {{es}}, but plugins can not access them directly: {{es}} enforces that certain files can only be read by its core code, while some other files can not be read or written at all. A plugin is always granted `read` access to the {{es}} config directory and `read_write` access to the temp directory; if the plugin requires to read, write or access additional files or directories, it must specify them via this entitlement.
+
+It is possible to specify 3 different types of file entitlement:
+  - `path` to specify an absolute path
+  - `relative_path` to specify a relative path. The path will be resolved via the `relative_to` field, which is used to qualify the relative path. It can be a specific {{es}} directory (`config` or `data`), or to the user home directory (`home`) (the home of the user running {{es}})
+  - `path_setting` to specify a path defined via an {{es}} setting. The path can be absolute or relative; in the latter case, the path will be resolved using the `basedir_if_relative` path (which can assume the same values as `relative_to`)
+
+Each of the 3 types has some additional fields:
+- `mode` (required): can be either `read` or `read_write`
+- `platform` (optional): if a path is specific to a platform. Can be `linux`, `macos` or `windows`. If not specified, the path is assumed to be valid on all platforms.
+- `exclusive`: access to this path is exclusive for this plugin; this means that other plugins will not be able to access to it, not even if they have an entitlement that would normally grant access to that path.
+
+Example:
+```yaml
+org.example.module: # or 'ALL-UNNAMED' if the plugin is non-modular
+  - files:
+    - path: "/absolute/path"
+      mode: read
+    - relative_path: "relative/file.txt"
+      relative_to: data
+      mode: read_write
+    - path_setting: setting.name
+      basedir_if_relative: data
+      mode: read
+```
+
+
+#### `write_system_properties`
```

Comment:
There's also `write_all_system_properties`, which is an external entitlement.

Reply:
But it's for modules only, it's not available to plugins